drivers/pppou.c

```diff
@@ -585,7 +592,9 @@ static int pppou_netifInit(struct netif *netif, char *cfg)
 		}
 		netif->flags &= ~NETIF_FLAG_UP;
 		ppp_set_netif_statuscallback(state->ppp, pppou_statusCallback);
-
+#if LWIP_DNS
+		ppp_set_usepeerdns(state->ppp, 1);
```

Comment:
You can add similar logic to the one used with the `ppp_set_default ()` route below, e.g. by using the `CFG_FLAG_NO_DNS` flag (configuration option`"nodns"`), this way DNS will be enabled by default but can be easily disabled in pppou configuration. Of course, between ifdefs `LWIP_DNS`.